scripts/generate_records/db_record_loaders.js

```diff
@@ -0,0 +1,266 @@
+// @ts-check
+
+/* eslint-disable no-await-in-loop */
+
+const Logger = require('@cumulus/logger');
+const {
+  CollectionPgModel,
+  ProviderPgModel,
+  fakeGranuleRecordFactory,
+  fakeFileRecordFactory,
+  fakeExecutionRecordFactory,
+  fakeRuleRecordFactory,
+  RulePgModel,
+  fakeCollectionRecordFactory,
+  fakeProviderRecordFactory,
+} = require('@cumulus/db');
+const { randomString } = require('@cumulus/common/test-utils');
+
+const log = new Logger({
+  sender: '@cumulus/generate_records',
+});
+
+/**
+ * @typedef {import('@cumulus/db').PostgresFile} PostgresFile
+ * @typedef {import('@cumulus/db').PostgresGranule} PostgresGranule
+ * @typedef {import('@cumulus/db').PostgresCollection} PostgresCollection
+ * @typedef {import('@cumulus/db').GranulesExecutionsPgModel} GranulesExecutionsPgModel
+ * @typedef {import('@cumulus/db').ExecutionPgModel} ExecutionPgModel
+ * @typedef {import('@cumulus/db').GranulePgModel} GranulePgModel
+ * @typedef {import('@cumulus/db').FilePgModel} FilePgModel
+ * @typedef {import('knex').Knex} Knex
+ * @typedef {{
+*   geModel: GranulesExecutionsPgModel,
+*   executionModel: ExecutionPgModel,
+*   granuleModel: GranulePgModel,
+*   fileModel: FilePgModel
+* }} ModelSet
+* @typedef {{
+*   name: string,
+*   version: string,
+* }} CollectionDetails
+*/
+/**
+ * upload executions corresponding to collection with collectionCumulusId
+ *
+ * @param {Knex} knex
+ * @param {number} collectionCumulusId
+ * @param {number} executionCount
+ * @param {ExecutionPgModel} model
+ * @param {boolean} swallowErrors
+ * @returns {Promise<Array<number>>} - cumulusId for each successfully uploaded execution
+ */
+const loadExecutions = async (
+  knex,
+  collectionCumulusId,
+  executionCount,
+  model,
+  swallowErrors = false
+) => {
+  const executionCumulusIds = [];
+  for (let i = 0; i < executionCount; i += 1) {
+    const execution = fakeExecutionRecordFactory({ collection_cumulus_id: collectionCumulusId });
+    try {
+      const [executionOutput] = await model.upsert(knex, execution);
```

Comment:
this is a great idea, I thought I couldn't do this because of potential unique-id conflicts, but having upped the length of random strings and wrapped inserts with a try catch this is dramatically faster, cleaner, errors are caught and also vanishingly rare